### PR TITLE
RMB-933 exclude self reference elements from URI encoding

### PR DESCRIPTION
**Pull request by grabsefx**
https://issues.folio.org/browse/RMB-933
*  exclude self reference elements from URI encoding

### CODE REVIEW

**Review by SerhiiNosko — APPROVED** [review]

File: domain-models-maven-plugin/src/main/java/org/folio/rest/tools/utils/SchemaDereferencer.java [hunk 1]
```diff
@@ -106,4 +106,8 @@ static String toFileUri(Path basePathFile, String file) {
     Path absolutePath = basePathFile.resolveSibling(file).normalize();
     return absolutePath.toUri().toString();
   }
+
+  private static boolean hasSelfReference(String refValue) {
+    return refValue.contains("#");
+  }
```
Comment (adamdickmeiss) [inline domain-models-maven-plugin/src/main/java/org/folio/rest/tools/utils/SchemaDereferencer.java:112]: Either check for first character to make this conform to the method name ; or rename the method to say hasUriFragment in which the method does what it says. In both cases, things should work. The hasUriScheme would also return true if fragment in a URL.

**Review by adamdickmeiss — CHANGES_REQUESTED** [review]

File: domain-models-maven-plugin/src/main/java/org/folio/rest/tools/utils/SchemaDereferencer.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/utils/SchemaDereferencer.java:112]: I prefer the former.

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Review by adamdickmeiss — APPROVED** [review]

**Comment by grabsefx** [comment]
@adamdickmeiss could you merge this PR? It seems I dont have permissisions to do that

**Comment by adamdickmeiss** [comment]
Thx for the contribution

**Pull request merged**